lib/http/2/compressor.rb

```diff
@@ -513,10 +503,9 @@ def string(buf)
         huffman = (buf.readbyte(0) & 0x80) == 0x80
         len = integer(buf, 7)
         str = buf.read(len)
-        str.bytesize == len or raise CompressionError.new("string too short")
-        huffman and str = Huffman.new.decode(Buffer.new(str))
-        str = str.force_encoding('utf-8')
-        str
+        str.bytesize == len || fail(CompressionError, 'string too short')
+        huffman && str = Huffman.new.decode(Buffer.new(str))
```

Comment:
These examples are all condition `and` action style.  I prefer `and`/`or`.
Probably we might want `if`/`unless` solution here:
- `raise ... unless str.bytesize == len`
- `str = Huffman.new(...) if huffman`

「condition `and` action」 vs 「action `if` condition」 is another question, but I don't want to dive into this argument here.

BTW, it's a surprise that `condition && var = value` works!  Ruby's parser seems to be too clever.
I expect it parses as `(condition && var) = value` and gives a syntax error.
`x = a && b` parses into ` x = (a && b)` but `a && b = x` becomes `(a) && ( b = x )`.
That's weird.
This behavior is Ruby-only and not consistent with Perl and some other. (AFAIK)